### PR TITLE
Terra dark mode case study

### LINKED ISSUE #6653: [Designcember][A11y][Fugu] Terra improves bounce rates and pages / session thanks to Dark Mode [12/18/2021]
* Proposed title: Terra improves bounce rates and pages / session thanks to Dark Mode
* Proposed by: drenzulli
* Related areas: Accessibility,Fugu/Capabilities
* Content type: Case Study (https://web.dev/handbook/case-study-guidelines/)
* Deadline: 11/30/2021
* Deadline explanation: This article is meant to be published for Designcember. I'll be OOO in December, so it would be great to have it uploaded to Web.dev by the end of November.
* [Draft](https://docs.google.com/presentation/d/1GjKMK5U46wZsG4uSlRtojWGhdKRMx85d7fOTh4mBwyo/edit)
* Content summary: A case study summarizing Terra's Dark Mode experiment, where they were able to measure the impact of providing a dark theme for users, leading to lower bounce rates and higher pages / session

### PR DESCRIPTION
**Pull request by demianrenzulli**
Fixes #6653

Changes proposed in this pull request:

- Terra Dark Mode case study.
- New profile for co-author Guilherme Moser De Souza. 

cc // @rachelandrew @andreban

### CODE REVIEW

File: src/site/content/en/blog/terra-dark-mode/index.md [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+## What is dark mode?
+
+Historically user interfaces in devices are displayed in "light mode", which usually means displaying black text on top of light interfaces. The alternative is "dark mode", with light text on a dark background, such as gray or black.
+
+Dark Mode has [benefits](https://web.dev/prefers-color-scheme/#why-dark-mode) for user experience. Some people prefer it for aesthetic or accessibility reasons. It has  other advantages, such as preserving battery life in devices. Users can express that they prefer dark mode via a setting in their devices, [which depends on the operating system](https://web.dev/prefers-color-scheme/#activating-dark-mode-in-the-operating-system). For example, the following screenshot shows what the **Dark Theme** configuration option looks like in devices that run **Android Q**:
+
+<figure class="w-figure">
+  {% Img src="image/admin/Yh6SEoWDK1SbqcGjlL6d.png", alt="Android Q dark mode settings", width="218", height="250" %}
+  <figcaption class="w-figcaption">Android&nbsp;Q dark theme settings</figcaption>
+</figure>
```
Comment (rachelandrew) [inline src/site/content/en/blog/terra-dark-mode/index.md:46]:
```suggestion
<figure>
  {% Img src="image/admin/Yh6SEoWDK1SbqcGjlL6d.png", alt="Android Q dark mode settings.", width="218", height="250" %}
  <figcaption>Android&nbsp;Q dark theme settings.</figcaption>
</figure>
```

File: src/site/content/en/blog/terra-dark-mode/index.md [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+
+@media (prefers-color-scheme: light) {
+  // apply a light theme
+}
+```
+
+{% BrowserCompat 'css.at-rules.media.prefers-color-scheme' %}
+
+{% Aside %}
+This article will only cover the technique of applying a custom dark theme, provided by the developer. Chrome 96 has introduced an [origin trial](https://developer.chrome.com/blog/auto-dark-theme/#sign-up-for-the-origin-trial) for "Auto Dark Themes" on Android, for which the browser applies an automatically generated dark theme to light themed sites, when the user has opted into dark themes in the operating system, without requiring the developer to provide styles for it. For more information about &"Chrome Auto Dark Mode", check out [this article](https://developer.chrome.com/blog/auto-dark-theme/).
```
Comment (rachelandrew) [inline src/site/content/en/blog/terra-dark-mode/index.md:63]:
```suggestion
This article will only cover the technique of applying a custom dark theme, provided by the developer. Chrome 96 has introduced an [origin trial](https://developer.chrome.com/blog/auto-dark-theme/#sign-up-for-the-origin-trial) for "Auto Dark Themes" on Android, for which the browser applies an automatically generated dark theme to light themed sites, when the user has opted into dark themes in the operating system, without requiring the developer to provide styles for it. For more information about "Chrome Auto Dark Mode", check out [this article](https://developer.chrome.com/blog/auto-dark-theme/).
```

File: src/site/content/en/blog/terra-dark-mode/index.md [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+
+Terra decided to give more control to the user: when they detect that they have the "prefer dark" setting turned on in their devices (via the media query), they show them a prompt to ask them if they want to navigate in "night mode". As long as the user doesn't deny the prompt (by clicking on the "Ignore" button), they honor the user's OS-setting, and apply a custom dark theme:
+
+<figure class="w-figure">
+  {% Img src="image/26V1DWN36MZr3mUo8ChSBlCpzp43/TRqfCAmBe025456JyX1b.png", alt="Screenshot of Terra's light theme prompting the user to accept dark mode.", width="266", height="146" %}
+  <figcaption class="w-figcaption">Terra shows a prompt to the user asking if they want to navigate in dark mode after detecting that they prefer dark in their devices.</figcaption>
+</figure>
+
+As a complement of this strategy they provide additional configuration options in the "settings" screen, where the user can decide if they explicitly prefer "light", "dark" or just rely on the underlying device settings:
```
Comment (rachelandrew) [inline src/site/content/en/blog/terra-dark-mode/index.md:118]:
```suggestion
As a complement of this strategy they provide additional configuration options in the "settings" screen, where the user can decide if they explicitly prefer "light", "dark", or want to rely on the underlying device settings.
```

File: src/site/content/en/blog/terra-dark-mode/index.md [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+<figure class="w-figure">
+  {% Img src="image/26V1DWN36MZr3mUo8ChSBlCpzp43/B7g0uvq2QB0eWVjnuMAl.png", alt="Screenshots of Terra's configuration screen to opt in and out of dark mode.", width="480", height="417" %}
+  <figcaption class="w-figcaption">Terra's themes configurations allow users to choose between "Dark" and "Light" themes or rely on the device's settings.</figcaption>
+</figure>
+
+This is how Terra's "Night Mode" looks like:
+
+<figure class="w-figure">
+  {% Img src="image/26V1DWN36MZr3mUo8ChSBlCpzp43/QRW06FYeMghUI8obAQWC.png", alt="Screenshot of Terra's dark theme.", width="286", height="468" %}
+  <figcaption class="w-figcaption">Therra's dark theme, "Night Mode".</figcaption>
```
Comment (rachelandrew) [inline src/site/content/en/blog/terra-dark-mode/index.md:129]:
```suggestion
  <figcaption>Terra's dark theme, "Night Mode".</figcaption>
```

File: src/site/content/en/blog/terra-dark-mode/index.md [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+
+## Measuring the impact of Terra's dark theme
+
+To measure the impact of the dark theme, Terra took the group of users that have the "Prefer Dark" setting turned on in their devices and compared engagement metrics when showing a "Light" vs. a "DarK" theme.
+Here are the results for mobile (Android) and desktop (Windows):
+
+{% Img src="image/26V1DWN36MZr3mUo8ChSBlCpzp43/eKI3Hj1ZXh2WxNPTOoay.png", alt="Terra stats after applying a dark theme showing how bounce rates and pages per session improved for users that prefer dark.", width="800", height="438" %}
+
+
+- **Mobile (Android):** While bounce rates remained similar, pages and sessions almost doubled (from 2.47 to 5.24) when users were exposed to a dark theme.
```
Comment (rachelandrew) [inline src/site/content/en/blog/terra-dark-mode/index.md:144]:
```suggestion
### Mobile (Android

While bounce rates remained similar, pages and sessions almost doubled (from 2.47 to 5.24) when users were exposed to a dark theme.
```

File: src/site/content/en/blog/terra-dark-mode/index.md [hunk 6]
```diff
@@ -0,0 +1,174 @@
...
+
+<div class="w-stats">
+  <div class="w-stat">
+    <p class="w-stat__figure">2<sub class="w-stat__sub">X</sub></p>
+    <p class="w-stat__desc">More pages per session</p>
+  </div>
+</div>
+
+- **Desktop (Windows):** Both metrics improved when showing a dark theme: bounce rates went from 27.25% to 10.82% and pages per session almost tripled (from 3.7 to 9.99).
```
Comment (rachelandrew) [inline src/site/content/en/blog/terra-dark-mode/index.md:153]:
```suggestion
### Desktop (Windows)

Both metrics improved when showing a dark theme: bounce rates went from 27.25% to 10.82% and pages per session almost tripled (from 3.7 to 9.99).
```

File: src/site/content/en/blog/terra-dark-mode/index.md [hunk 7]
```diff
@@ -0,0 +1,174 @@
+---
+layout: post-old
```
Comment (rachelandrew) [inline src/site/content/en/blog/terra-dark-mode/index.md:2]:
```suggestion
layout: post
```

File: src/site/content/en/blog/terra-dark-mode/index.md [hunk 8]
```diff
@@ -0,0 +1,174 @@
...
+<div class="w-stats">
+  <div class="w-stat">
+    <p class="w-stat__figure">60<sub class="w-stat__sub">%</sub></p>
+    <p class="w-stat__desc">Reduction in Bounce Rates</p>
+  </div>
+  <div class="w-stat">
+    <p class="w-stat__figure">170<sub class="w-stat__sub">%</sub></p>
+    <p class="w-stat__desc">More pages per session</p>
+  </div>
+</div>
```
Comment (rachelandrew) [inline src/site/content/en/blog/terra-dark-mode/index.md:35]:
```suggestion
<ul class="stats">
  <div class="stats__item">
    <p class="stats__figure">60<sub>%</sub></p>
    <p>Reduction in Bounce Rates</p>
  </div>
  <div class="stats__item">
    <p class="stats__figure">170<sub>%</sub></p>
    <p>More pages per session</p>
  </div>
</ul>
```

File: src/site/content/en/blog/terra-dark-mode/index.md [hunk 9]
```diff
@@ -0,0 +1,174 @@
...
+
+## What is dark mode?
+
+Historically user interfaces in devices are displayed in "light mode", which usually means displaying black text on top of light interfaces. The alternative is "dark mode", with light text on a dark background, such as gray or black.
+
+Dark Mode has [benefits](https://web.dev/prefers-color-scheme/#why-dark-mode) for user experience. Some people prefer it for aesthetic or accessibility reasons. It has  other advantages, such as preserving battery life in devices. Users can express that they prefer dark mode via a setting in their devices, [which depends on the operating system](https://web.dev/prefers-color-scheme/#activating-dark-mode-in-the-operating-system). For example, the following screenshot shows what the **Dark Theme** configuration option looks like in devices that run **Android Q**:
```
Comment (rachelandrew) [inline src/site/content/en/blog/terra-dark-mode/index.md:41]:
```suggestion
Dark Mode has [benefits](/prefers-color-scheme/#why-dark-mode) for user experience. Some people prefer it for aesthetic or accessibility reasons. It has  other advantages, such as preserving battery life in devices. Users can express that they prefer dark mode via a setting in their devices, [which depends on the operating system](/prefers-color-scheme/#activating-dark-mode-in-the-operating-system). For example, the following screenshot shows what the **Dark Theme** configuration option looks like in devices that run **Android Q**:
```

File: src/site/content/en/blog/terra-dark-mode/index.md [hunk 10]
```diff
@@ -0,0 +1,174 @@
...
+## Implementing a dark theme
+
+Most websites implement a dark theme by using the simple strategy shown previously of listening to user's configuration changes via the [`prefers-color-scheme`](https://drafts.csswg.org/mediaqueries-5/#prefers-color-scheme) media query and changing styles based on that.
+
+Terra decided to give more control to the user: when they detect that they have the "prefer dark" setting turned on in their devices (via the media query), they show them a prompt to ask them if they want to navigate in "night mode". As long as the user doesn't deny the prompt (by clicking on the "Ignore" button), they honor the user's OS-setting, and apply a custom dark theme:
+
+<figure class="w-figure">
+  {% Img src="image/26V1DWN36MZr3mUo8ChSBlCpzp43/TRqfCAmBe025456JyX1b.png", alt="Screenshot of Terra's light theme prompting the user to accept dark mode.", width="266", height="146" %}
+  <figcaption class="w-figcaption">Terra shows a prompt to the user asking if they want to navigate in dark mode after detecting that they prefer dark in their devices.</figcaption>
+</figure>
```
Comment (rachelandrew) [inline src/site/content/en/blog/terra-dark-mode/index.md:116]:
```suggestion
<figure>
  {% Img src="image/26V1DWN36MZr3mUo8ChSBlCpzp43/TRqfCAmBe025456JyX1b.png", alt="Screenshot of Terra's light theme prompting the user to accept dark mode.", width="266", height="146" %}
  <figcaption>Terra shows a prompt to the user asking if they want to navigate in dark mode after detecting that they prefer dark in their devices.</figcaption>
</figure>
```

File: src/site/content/en/blog/terra-dark-mode/index.md [hunk 11]
```diff
@@ -0,0 +1,174 @@
...
+
+As a complement of this strategy they provide additional configuration options in the "settings" screen, where the user can decide if they explicitly prefer "light", "dark" or just rely on the underlying device settings:
+
+<figure class="w-figure">
+  {% Img src="image/26V1DWN36MZr3mUo8ChSBlCpzp43/B7g0uvq2QB0eWVjnuMAl.png", alt="Screenshots of Terra's configuration screen to opt in and out of dark mode.", width="480", height="417" %}
+  <figcaption class="w-figcaption">Terra's themes configurations allow users to choose between "Dark" and "Light" themes or rely on the device's settings.</figcaption>
+</figure>
+
+This is how Terra's "Night Mode" looks like:
```
Comment (rachelandrew) [inline src/site/content/en/blog/terra-dark-mode/index.md:125]:
```suggestion
<figure>
  {% Img src="image/26V1DWN36MZr3mUo8ChSBlCpzp43/B7g0uvq2QB0eWVjnuMAl.png", alt="Screenshots of Terra's configuration screen to opt in and out of dark mode.", width="480", height="417" %}
  <figcaption>Terra's themes configurations allow users to choose between "Dark" and "Light" themes or rely on the device's settings.</figcaption>
</figure>

This is how Terra's "Night Mode" looks like:
```

File: src/site/content/en/blog/terra-dark-mode/index.md [hunk 12]
```diff
@@ -0,0 +1,174 @@
...
+
+<figure class="w-figure">
```
Comment (rachelandrew) [inline src/site/content/en/blog/terra-dark-mode/index.md:127]:
```suggestion
<figure>
```

File: src/site/content/en/blog/terra-dark-mode/index.md [hunk 13]
```diff
@@ -0,0 +1,174 @@
...
+
+
+- **Mobile (Android):** While bounce rates remained similar, pages and sessions almost doubled (from 2.47 to 5.24) when users were exposed to a dark theme.
+
+<div class="w-stats">
+  <div class="w-stat">
+    <p class="w-stat__figure">2<sub class="w-stat__sub">X</sub></p>
+    <p class="w-stat__desc">More pages per session</p>
+  </div>
+</div>
```
Comment (rachelandrew) [inline src/site/content/en/blog/terra-dark-mode/index.md:151]:
```suggestion
<ul class="stats">
  <div class="stats__item">
    <p class="stats__figure">2<sub>X</sub></p>
    <p>More pages per session</p>
  </div>
</ul>
```

File: src/site/content/en/blog/terra-dark-mode/index.md [hunk 14]
```diff
@@ -0,0 +1,174 @@
...
+<div class="w-stats">
+  <div class="w-stat">
+    <p class="w-stat__figure">60<sub class="w-stat__sub">%</sub></p>
+    <p class="w-stat__desc">Reduction in Bounce Rates</p>
+  </div>
+  <div class="w-stat">
+    <p class="w-stat__figure">170<sub class="w-stat__sub">%</sub></p>
+    <p class="w-stat__desc">More pages per session</p>
+  </div>
+</div>
```
Comment (rachelandrew) [inline src/site/content/en/blog/terra-dark-mode/index.md:164]:
```suggestion
<ul class="stats">
  <div class="stats__item">
    <p class="stats_figure"><sub>60%</sub></p>
    <p>Reduction in Bounce Rates</p>
  </div>
  <div class="stats__item">
    <p class="stats__figure">170<sub>%</sub></p>
    <p>More pages per session</p>
  </div>
</ul>
```

**Review by rachelandrew — CHANGES_REQUESTED** [review]
See comments inline, this PR crossed over the component code change so I've had a go at updating to the new components too so this one can be in the new layout.

**Comment by demianrenzulli** [comment]
Thanks for reviewing this PR @rachelandrew! I just committed all your suggestions and run it locally. It seems that the components and images look good.

Please, let me know if you need anything else from me.

**Comment by demianrenzulli** [comment]
Update:

@rachelandrew: I just ran the site locally again, after pulling the latest changes from `main` and merging them back to my branch. I see that the "stats" component is displayed as plain text and images appear aligned to the left, instead of centered. Would it be possible for you to double-check on your side what might be wrong?

@jpmedley: I just committed the changes for the suggestions you have made. I still need to work on the tables. I understand that you are referring to the [tables component](https://web.dev/handbook/web-dev-components/#tables), but if you have any article example to share that's using them properly that would help.

**Review by andreban — APPROVED** [review]
LGTM (from a Dark Theme perspective)

**Review by rachelandrew — APPROVED** [review]

**Comment by demianrenzulli** [comment]
Hi team, thanks again for taking a look at this PR and leaving comments. It seems like we have all the approvals now.

I see that the [preview](https://deploy-preview-6898--web-dev-staging.netlify.app/terra-dark-mode/) still shows the "stats" components broken and the images are aligned to left instead of centered.

@rachelandrew: I recall you've mentioned that this will be fixed at the time of launching this (Dec 18th), is that correct? 
Even if that's the case, Is there anything we can do to be able to preview the content with the components fixed before the launch date? I'd like to share a final version with the partner, so we make sure that everything looks good.

Thanks in advance.

File: src/site/content/en/blog/terra-dark-mode/index.md [hunk 15]
```diff
@@ -0,0 +1,275 @@
...
+        <td>10.82%</td>
+        <td>9.99</td>
+      </tr>
+    </tbody>
+  </table>
+</div>
+
+<ul class="stats">
+  <div class="stats__item">
+    <p class="stats_figure"><sub>60%</sub></p>
```
Comment (rachelandrew) [inline src/site/content/en/blog/terra-dark-mode/index.md:258]:
```suggestion
    <p class="stats_figure">60<sub>%</sub></p>
```

File: src/site/content/en/blog/terra-dark-mode/index.md [hunk 16]
```diff
@@ -0,0 +1,275 @@
...
+        <td>10.82%</td>
+        <td>9.99</td>
+      </tr>
+    </tbody>
+  </table>
+</div>
+
+<ul class="stats">
+  <div class="stats__item">
+    <p class="stats_figure">60<sub>%</sub></p>
```
Comment (rachelandrew) [inline src/site/content/en/blog/terra-dark-mode/index.md:258]:
```suggestion
    <p class="stats__figure">60<sub>%</sub></p>
```

**Comment by demianrenzulli** [comment]
Just in case it's easier to see it here, or anyone in Web.dev side can help: 

The images are not being sized correctly in the production version of this article, due to the usage of the new component: https://web.dev/terra-dark-mode/.

The [tweet has been published earlier today](https://twitter.com/ChromiumDev/status/1472237217858957323), so If there's anything we can do to help, please, let us know.

cc // @mobtec

**Pull request merged**